src/main/java/uk/gov/companieshouse/api/strikeoffobjections/file/FileTransferApiClient.java

```diff
@@ -167,15 +174,76 @@ public FileTransferApiClientResponse delete(String requestId, String fileId) {
      * @param httpServletResponse The HttpServletResponse to stream the file to
      * @return FileTransferApiClientResponse containing the http status
      */
-    public FileTransferApiClientResponse download(String fileId, HttpServletResponse httpServletResponse) throws IOException {
-        // TODO OBJ-200 replace this dummy response with the implementation.
-        FileTransferApiClientResponse fileTransferApiClientResponse = new FileTransferApiClientResponse();
-        fileTransferApiClientResponse.setFileId(fileId);
-        fileTransferApiClientResponse.setHttpStatus(HttpStatus.OK);
+    public FileTransferApiClientResponse download(String requestId, String fileId, HttpServletResponse httpServletResponse) {
+        String downloadUri = String.format(DOWNLOAD_URI, fileTransferApiURL, fileId);
+
+        return makeApiCall(
+                requestId,
+                //FileTransferOperation
+                () -> restTemplate.execute(
+                        downloadUri,
+                        HttpMethod.GET,
+                        this::handleRequestCallback,
+                        clientHttpResponse -> getClientHttpResponse(httpServletResponse, clientHttpResponse)
+                        ),
+                clientHttpResponse -> getFileTransferApiClientResponse(requestId, clientHttpResponse)
+        );
+    }
+
+    private boolean isFileClean(String requestId, String fileId) {
+        String getUrl = String.format(FILE_TRANSFER_URI_TEMPLATE, fileTransferApiURL, fileId);
+
+        FileTransferApiClientResponse fileDataResponse = makeApiCall(
+                requestId,
+                () -> {
+                    HttpEntity<Void> request = new HttpEntity<>(createApiKeyHeader());
+                    return restTemplate.exchange(getUrl, HttpMethod.DELETE, request, FileTransferApiResponse.class);
+                },
+                responseEntity -> getFileTransferApiClientResponse(requestId, responseEntity)
+        );
+
+        return fileDataResponse.getAvStatus().equals("clean");
+
+    }
+
+    private void handleRequestCallback(ClientHttpRequest requestCallback) {
+        requestCallback.getHeaders().setAccept(Collections.singletonList(MediaType.APPLICATION_OCTET_STREAM));
+        requestCallback.getHeaders().add(HEADER_API_KEY, fileTransferApiKey);
+    }
 
-        InputStream inputStream = new ByteArrayInputStream("This is a test".getBytes());
+    private ClientHttpResponse getClientHttpResponse(HttpServletResponse httpServletResponse, ClientHttpResponse clientHttpResponse) throws IOException {
```

Comment:
this method looks like it is doing the copying, the getClientHttpResponse name doesn't indicate that any copying will take place. Also we are passing a ClientHttpResponse in to the getClientHttpResponse which is confusing.

Reply:
Perhaps "copyClientHttpDataToServletResponse" would be better

Reply:
Changed name.